pennylane/transforms/decompositions/single_qubit_unitary.py

```diff
@@ -26,29 +26,68 @@ def _convert_to_su2(U, return_global_phase=False):
-        return_global_phase (bool): If `True`, the return will include
-        the global phase. If `False`, only the :math:`SU(2)` representative
-        is returned.
+            of shape :math:`n \times 2 \times 2` and unitary for any positive integer n.
+        return_global_phase (bool): If `True`, the return will include the global phase.
+            If `False`, only the :math:`SU(2)` representation is returned.
 
     Returns:
         array[complex]: A :math:`n \times 2 \times 2` matrix in :math:`SU(2)` that is
-        equivalent to U up to a global phase. If ``return_global_phase=True``,
-        a 2-element tuple is returned, with the first element being the
-        :math:`SU(2)` equivalent and the second, the global phase.
+            equivalent to U up to a global phase. If ``return_global_phase=True``, a
+            2-element tuple is returned, with the first element being the :math:`SU(2)`
+            equivalent and the second, the global phase.
     """
+
     # Compute the determinants
     U = qml.math.cast(U, "complex128")
-    dets = math.linalg.det(U)
-    exp_angles = math.angle(dets) / 2
-    U_SU2 = math.cast_like(U, dets) * math.exp(-1j * math.cast_like(exp_angles, 1j))[:, None, None]
+    determinants = math.linalg.det(U)
+    phase = math.angle(determinants) / 2
+    U = math.cast_like(U, determinants) * math.exp(-1j * math.cast_like(phase, 1j))[:, None, None]
+
+    return (U, phase) if return_global_phase else U
+
+
+def _zyz_get_rotation_angles(U):
+    r"""Computes the rotation angles :math:`\phi`, :math:`\theta`, :math:`\omega`
+    for a unitary :math:`U` that is :math:`SU(2)`
+
+    Args:
+        U (array[complex]): A matrix that is :math:`SU(2)`
 
-    return (U_SU2, exp_angles) if return_global_phase else U_SU2
+    Returns:
+        tuple[array[float]]: A tuple containing the rotation angles
+            :math:`\phi`, :math:`\theta`, :math:`\omega`
 
+    """
 
-def _rot_decomposition(U, wire):
-    r"""Recover the decomposition of a single-qubit matrix :math:`U` in terms of
-    elementary operations.
+    # For batched U or single U with non-zero off-diagonal, compute the
+    # normal decomposition instead
+    off_diagonal_elements = math.clip(math.abs(U[:, 0, 1]), 0, 1)
+    thetas = 2 * math.arcsin(off_diagonal_elements)
+
+    # Compute phi and omega from the angles of the top row; use atan2 to keep
+    # the angle within -np.pi and np.pi, and add very small value to the real
+    # part to avoid division by zero.
+    epsilon = 1e-64
+    angles_U00 = math.arctan2(
+        math.imag(U[:, 0, 0]),
+        math.real(U[:, 0, 0]) + epsilon,
+    )
+    angles_U10 = math.arctan2(
+        math.imag(U[:, 1, 0]),
+        math.real(U[:, 1, 0]) + epsilon,
+    )
+
+    phis = -angles_U10 - angles_U00
+    omegas = angles_U10 - angles_U00
+
+    phis, thetas, omegas = map(math.squeeze, [phis, thetas, omegas])
+
+    return phis, thetas, omegas
```

Comment:
If I am not wrong you can also add the normalization of angle in the helper function.